### PR TITLE
Editor: Refactored Background/Environment code

### PR DESCRIPTION
**Pull request by mrdoob**
Cleaned up the "none" background behaviour, removed the `CubeTexture` background (for now) and made `Environment` opt-in.

Screen Shot 2020-06-19 at 9 53 34 PM

### CODE REVIEW

File: editor/js/Sidebar.Scene.js [hunk 1]
```diff
@@ -376,29 +368,37 @@ function SidebarScene( editor ) {
 				backgroundType.setValue( "Color" );
 				backgroundColor.setHexValue( scene.background.getHex() );
 				backgroundTexture.setValue( null );
-				backgroundCubeTexture.setValue( null );
-				backgroundEquirectTexture.setValue( null );
+				backgroundEquirectangularTexture.setValue( null );
 
-			} else if ( scene.background.isTexture && ! scene.background.isPmremTexture ) {
+			} else if ( scene.background.isEquirectangularTexture ) {
```
Comment (Mugen87) [inline editor/js/Sidebar.Scene.js:373]: Where does `isEquirectangularTexture` comes from? Is this property added in a different commit?

**Comment by Mugen87** [comment]
BTW: Nice refactoring. UX should be better after this change (although I'm a bit sad that handling cube textures was removed 😉 ).

File: editor/js/Sidebar.Scene.js (hunk 1 above)
Comment (mrdoob) [reply editor/js/Sidebar.Scene.js:373]: Hmm... Yeah, still figuring that out. I should comment that code to avoid confusion.

**Comment by mrdoob** [comment]
Yeah, sorry about that!

Now that we have more functions to convert from equirect to cubemap, it's starting to feel like cubemap is more of an internal representation rather than a workflow.

I'm still undecided about it though.

File: editor/js/Sidebar.Scene.js (hunk 1 above)
Comment (mrdoob) [reply editor/js/Sidebar.Scene.js:373]: Done. https://github.com/mrdoob/three.js/commit/a36f902f6f59de4d0c717bf3d4be9947667c7895

**Pull request merged**